test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs

```diff
@@ -0,0 +1,209 @@
+        {
+            // Arrange
+            var codeDocument = CreateDocument(@"
+@inject PropertyType PropertyName
+@inject PropertyType2 PropertyName
+");
+
+            var engine = CreateEngine();
+            var pass = new InjectDirective.Pass()
+            {
+                Engine = engine,
+            };
+
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            pass.Execute(codeDocument, irDocument);
+
+            // Assert
+            var @class = FindClassNode(irDocument);
+            Assert.NotNull(@class);
+            Assert.Equal(2, @class.Children.Count);
+
+            var statement = Assert.IsType<CSharpStatementIRNode>(@class.Children[1]);
+            Assert.Equal(
+                "[Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]" + Environment.NewLine +
+                "public PropertyType2 PropertyName { get; private set; }",
+                statement.Content);
+        }
+
+        [Fact]
+        public void InjectDirectivePass_Execute_ExpandsTModel_WithDynamic()
+        {
+            // Arrange
+            var codeDocument = CreateDocument(@"
+@inject PropertyType<TModel> PropertyName
+");
+
+            var engine = CreateEngine();
+            var pass = new InjectDirective.Pass()
+            {
+                Engine = engine,
+            };
+
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            pass.Execute(codeDocument, irDocument);
+
+            // Assert
+            var @class = FindClassNode(irDocument);
+            Assert.NotNull(@class);
+            Assert.Equal(2, @class.Children.Count);
+
+            var statement = Assert.IsType<CSharpStatementIRNode>(@class.Children[1]);
+            Assert.Equal(
+                "[Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]" + Environment.NewLine +
+                "public PropertyType<dynamic> PropertyName { get; private set; }",
+                statement.Content);
+        }
+
+        [Fact]
+        public void InjectDirectivePass_Execute_ExpandsTModel_WithModelType()
+        {
+            // Arrange
+            var codeDocument = CreateDocument(@"
+@inject PropertyType<TModel> PropertyName
+@model ModelType
+");
+
+            var engine = CreateEngine();
+            var pass = new InjectDirective.Pass()
+            {
+                Engine = engine,
+            };
+
+            var irDocument = CreateIRDocument(engine, codeDocument);
+
+            // Act
+            pass.Execute(codeDocument, irDocument);
+
+            // Assert
+            var @class = FindClassNode(irDocument);
+            Assert.NotNull(@class);
+            Assert.Equal(2, @class.Children.Count);
+
+            var statement = Assert.IsType<CSharpStatementIRNode>(@class.Children[1]);
+            Assert.Equal(
+                "[Microsoft.AspNetCore.Mvc.Razor.Internal.RazorInjectAttribute]" + Environment.NewLine +
```

Comment:
This won't affect any designtime stuff, it's in generated code that no one will see.